### DISCUSSION TITLE
Download all videos in a playlist

### DISCUSSION THREAD

**Discussion by akilou** [Tzahi12345/YoutubeDL-Material#342] [Ideas] [answered]
"regular" youtube-dl (when using it in Terminal) allows you to input a YouTube playlist ID (after the "&list=" in the url) and it downloads all videos in the playlist. And similarly extracts just the audio, if that's what you want (which I usually do).

It seems that youtubedl-material doesn't do this. When I input a list id, nothing happens. When I input the full url, it downloads only the first video (or audio) in the list and stops.

Does youtubedl-material support YouTube playlists and I'm doing something wrong? Otherwise, can you add this functionality?

Thank you.

**Comment by Tzahi12345** [comment, marked as answer]
Sorry for the late response, which version are you on? Have you tried running the latest `nightly`?

**Reply by akilou** [reply]
Sorry, I'm new to Docker, so I'm not sure how to check which version of a container I'm running. I'm running Docker on my Synology NAS in DSM, if that matters. [This page](https://github.com/Tzahi12345/YoutubeDL-Material/releases/tag/v4.2) says that v4.2 is the latest and that it was released December 24, 2020. I installed YoutubeDL-Material after that date, so it's reasonable to assume I'm running 4.2. It's labeled as `tzahi12345/youtubedl-material:latest`. Would the `:latest` be `:nightly` if I were running that? How do I choose to install nightly rather than whatever I'm running now (if indeed it's `latest` and v4.2)?

**Reply by Tzahi12345** [reply]
Ah I feel so bad leaving you hanging, I've been terribly busy so forgive me for the delay (again)

Latest is v4.2, it's just a way to keep up with the most recent stable release. To run on nightly just change `latest` to `nightly` and run `docker-compose pull` to grab the latest version. Your files and everything will still be there, so no worries about updating. And you can always revert back to v4.2 if you don't like the `nightly`.

I just asked in case this was an issue in the `latest`/`4.2` tags, and has been fixed in `nightly` (there's been quite a few updates since the last release)

**Reply by akilou** [reply]
Don't apologize, you're providing me a free service!

I'll play around with it this weekend to run the nightly build. Thanks so much for the help.

**Reply by akilou** [reply]
Hi. I was able to install the `nightly` build and I'm experiencing the same behavior as `latest` when trying to download a playlist. I tried troubleshooting a bit and this is what I found:

There are basically 3 variables that I can control: Multi-download on/off; audio only on/off; entire url or just after "&list=".

For both multi-download and audio only, whether it's on or off, it works as expected. The playlist variable is where the issues is.

**If I paste in the whole url**, it just downloads the first video in the playlist and then stops. If audio only is checked, it downloads audio only, if multi-download is selected it refrains from playing, as expected.

**If I paste in just List ID**, nothing happens at all.

Let me know if I'm doing something wrong on my end or if you need help recreating the behavior, how else I can help.

**Reply by Tzahi12345** [reply]
Can you share the playlist URL you are trying to input? Perhaps it's not an acceptable one for youtube-dl

What I know works is https://www.youtube.com/playlist?list=<playlist_id>

**Reply by akilou** [reply]
a ha! That format works! I was just pasting in the url from the first video after clicking on a playlist. It's formatted like this: `https://www.youtube.com/watch?v=<video id>&list=<playlist id>`. After copying your format and pasting in just the playlist id at the end worked like it should. Thank you so much!

**Reply by Tzahi12345** [reply]
Awesome, great to hear! youtube-dl is a bit finnicky with formatting on YT URLs but it's actually even worse in sites like Twitch
